src/error.ts

```diff
@@ -136,23 +135,31 @@ export class MongoServerError extends MongoError {
   code?: number;
   codeName?: string;
   writeConcernError?: Document;
+  errInfo?: Document;
+  ok?: number;
+  topologyVersion?: TopologyVersion;
+  [key: string]: any;
 
-  constructor(message: Error | ErrorDescription) {
-    if (message instanceof Error) {
-      super(message);
-    } else {
-      super(message.message || message.errmsg || message.$err || 'n/a');
-      if (message.errorLabels) {
-        this[kErrorLabels] = new Set(message.errorLabels);
-      }
-
-      for (const name in message) {
-        if (name === 'errorLabels' || name === 'errmsg' || name === 'message') {
-          continue;
-        }
-
-        (this as any)[name] = message[name];
-      }
+  constructor(message: ErrorDescription) {
```

Comment:
I honestly don't think that the changes below improve readability, and there are extra steps added that actually make it harder to follow the logic; originally, we were just iterating over all props returned from the server and assigning them to the error as its own properties; there is no complexity in maintaining that. 

The only exceptions where we don't directly read and assign from the server error are `message, errmsg, errorLabels`. So, what is the benefit from explicitly extracting, for example, `message.code` ahead of time, putting it into a set, and then excluding it from being set in the loop in L161? 

Moreover, setting these properties and then dynamically generating the props to filter is introducing unnecessary inefficiency: the set of properties to exclude doesn't change from error to error, we don't need to compute it each time.

If you feel that strongly about the triple `||` statement, you could create a constant regex declaration outside this class `const serverPropsToExclude = /^errorLabels$|^errmsg$|^message$/` and then set the condition to `if (serverPropsToExclude.test(name)) { continue; }`

Reply:
> setting these properties and then dynamically generating the props to filter

I agree here, we could hoist this set of keys to a `KNOWN_SERVER_ERROR_PROPS` and it'll be a one time construction. I think that could go a long way to making the logic clearer around our intention here.

> So, what is the benefit from explicitly extracting, for example, message.code ahead of time

Nothing functionally, entirely from a dev perspective this hints to me what's expected on the MongoServerError class, maybe it is enough to keep the property declarations and their types at the top of the class now, but having explicit line by line assignment for these "well known" keys helps callout what is on the class. 

I think this might be the result of my bias perspective of, before this ticket, MongoServerError was a mystery box. It still is, but with intention, the idea is for it to inherit anything the server sends us as an error. I recommended we take this as far as we could in the direction of figuring what was actually defined on this class. 

I would be fine with removing the explicit assignment but keeping just the type defs at the top (along with `[key: string]: any` of course)

Reply:
I don't think we need a `KNOWN_SERVER_ERROR_PROPS` I think marking the corresponding block of the class prop type defs with a comment saying that they are known server error props would accomplish the same thing (regarding communicating intent) without having to explicitly write them out in actual code, too. I think it's only the 3 that are being excluded that are worth hoisting up, if we do this at all.